tests/sample/test_sample_models.py

```diff
@@ -0,0 +1,149 @@
+        self.assertEqual(new_sample.participant_id,p.kf_id)
+
+    
+    def test_sample_participant_relation(self):
+        participant_id, sample_id= self.create_participant_sample()
+        s = Sample.query.filter_by(external_id=sample_id).one_or_none()
+        p = Participant.query.filter_by(external_id=participant_id).one_or_none()
+        self.assertEqual(p.kf_id,s.participant_id)
+        self.assertEqual(p.sample.external_id,s.external_id)
+
+
+    def test_find_sampe(self):
+        participant_id, sample_id= self.create_participant_sample()
+        
+        # get sample
+        s= Sample.query.filter_by(external_id=sample_id).one_or_none()
+        self.assertEqual(s.external_id, "Test_Sample_0")
+        self.assertEqual(s.tissue_type, "Normal")
+        self.assertEqual(s.composition,"Test_comp_0")
+        self.assertEqual(s.external_id,sample_id)
+
+    def test_update_sample(self):
+        participant_id, sample_id= self.create_participant_sample()
+         # get sample
+        s= Sample.query.filter_by(external_id=sample_id).one_or_none()
+
+        s.tissue_type ="Tumor"
+          # get sample
+        s= Sample.query.filter_by(external_id=sample_id).one_or_none()
+        self.assertEqual(s.tissue_type, 'Tumor')
+        self.assertEqual(s.external_id,sample_id)
+
+    def test_delete_sample(self):
+        participant_id, sample_id= self.create_participant_sample()
+        #get Sample
+        s= Sample.query.filter_by(external_id=sample_id).one_or_none()
+        
+        #Delete Sample
+        db.session.delete(s)
+        db.session.commit()
+
+        s= Sample.query.filter_by(external_id=sample_id).one_or_none()
+        self.assertIs(s,None)
+
+        #p= Participant.query.filter_by(external_id=sample_id).one_or_none()
+        self.assertIsNone(Participant.query.filter_by(external_id=sample_id).one_or_none())
+
+    def test_delete_sample_participant(self):
+
+        participant_id, sample_id= self.create_participant_sample()
+
+        #Delete Participant
+        p = Participant.query.first()
+        db.session.delete(p)
+        db.session.commit()
+
+        s= Sample.query.filter_by(external_id=sample_id).one_or_none()
+        self.assertIs(s,None)
+ 
+        self.assertIsNone(Participant.query.filter_by(external_id=sample_id).one_or_none())
+
+    def test_not_null_constraint(self):
+
+    
+        """
+        Test sample cannot be created with out required parameters such as participant_id
+    
+        """    
+
+        # Create Sample
+        sample_id = "Test_Sample_0"
+    
+        # With Missing Kf_id
+        s = Sample(external_id='Test_Sample_0',tissue_type='Normal',composition='Test_comp_0')
+
+        #Add Sample to db
+        self.assertRaises(IntegrityError, db.session.add(s))
+
+    def test_foreign_key_constraint(self):
+
+    
+        """
+        Test sample cannot be created with out required parameters such as participant_id
```

Comment:
Maybe change the docstring to `Test sample cannot be created with an existing/valid Participant`